### DISCUSSION TITLE
Internationalized Routing + custom slug per language

### DISCUSSION THREAD

**Discussion by krystianjj** [vercel/next.js#22092] [Help]
When using a new feautre `Internationalized Routing` you can have sites like **example.com/en/about-us** and **example.com/de/about-us** but we know SEO like keyword in slug. It is possible to have different slug per lang. 
For example:
example.com/en/about-us
example.com/de/uber-uns

Now I am dealing with this by copying pages and use different name, but this is not good because I repeat the code.

**Comment by benweier** [comment]
You should definitely be able to use a unique slug per language, but it may impact the way you structure your data source (e.g. unique slugs per language in your data source). The slug should work as the identifier for your page so you can use a different value for this ID to access different "pages". Just take note of the challenges this may pose to switching languages if the page slug isn't the same between locales.

Adapting the example [from the docs](https://nextjs.org/docs/advanced-features/i18n-routing):

```
// pages/blog/[slug].js
export const getStaticPaths = ({ locales }) => {
  return {
    paths: [
      { params: { slug: 'post-1-for-english' }, locale: 'en-US' },
      { params: { slug: 'post-1-with-fr-keywords' }, locale: 'fr' },
    ],
    fallback: true,
  }
}
```

**Reply by krystianjj** [reply]
Thank you for answer @benweier. This solution works when we use dynamic slug, but how do that for pages like `about-company` or `service-name` without dynamic [slug] ?

**Reply by ernestrudziec** [reply]
Well, bump.

I need this almost in every project. Looks like a key-feature in modern applications.

For my purposes i just created a file [...params] and created my own routing in this file but that's probably not the best approach.
It worked, but man - that's a mess.

**Reply by esharmony** [reply]
I have the same issue, also wish to translate a non-dynamic slug, I am unsure how to use [...params] and create my own routing, would it be possible to explain this to me or point me to a bit of documentation that could help?
